crates/relayer/src/chain/ckb4ibc/utils.rs

```diff
@@ -158,26 +169,50 @@ pub fn get_packet_lock_script(converter: &impl MsgToTxConverter, args: Vec<u8>)
         .build()
 }
 
-pub fn get_search_key(script: Script) -> SearchKey {
+pub fn get_prefix_search_key(script: Script) -> SearchKey {
     SearchKey {
         script: script.into(),
         script_type: ScriptType::Lock,
         filter: None,
         with_data: Some(true),
         group_by_transaction: None,
+        script_search_mode: Some(ScriptSearchMode::Prefix),
     }
 }
 
-pub fn get_connection_capacity() -> Capacity {
-    Capacity::bytes(CONNECTION_CELL_CAPACITY as usize).unwrap()
-}
-
-pub fn get_channel_capacity() -> Capacity {
-    Capacity::bytes(CHANNEL_CELL_CAPACITY as usize).unwrap()
-}
-
-pub fn get_packet_capacity() -> Capacity {
-    Capacity::bytes(PACKET_CELL_CAPACITY as usize).unwrap()
+pub fn get_search_key_with_sudt(
+    script: Script,
+    symbol: &str,
+    network: NetworkType,
+) -> Result<SearchKey, Error> {
+    let sudt_code_hash = match network {
+        NetworkType::Mainnet => SUDT_CODE_HASH_MAINNET,
+        NetworkType::Testnet => SUDT_CODE_HASH_TESTNET,
+        _ => {
+            return Err(Error::other_error(format!(
+                "unsupported network: {network}"
+            )))
+        }
+    };
+    let owner_lockhash =
+        H256::from_str(symbol).map_err(|_| Error::other_error("invalid sUDT symbol".to_owned()))?;
+    let sudt_script = Script::new_builder()
+        .code_hash(sudt_code_hash.pack())
+        .hash_type(ScriptHashType::Type.into())
+        .args(owner_lockhash.as_bytes().to_vec().pack())
+        .build();
+    let filter = SearchKeyFilter {
+        script: Some(sudt_script.into()),
+        ..Default::default()
```

Comment:
``` rust
script_search_mode: Some(ScriptSearchMode::Exact),
```

Reply:
This is `SearchKeyFilter`, it doesn't have an exact mode, but there's `script_len_range`.

Reply:
mark the issue, `script_len_range` should be considered in this case?

Reply:
IMO we should always verify the cell's format to eliminate the factor.

Reply:
got it, it's out of my experience